src/domain/core/src/services/verification_service.rs

```diff
@@ -55,6 +55,12 @@ pub enum VerificationResult {
     Valid,
 }
 
+#[derive(Debug)]
+pub struct VerificationMultiResult {
```

Comment:
Why do you call it "multi"? It actually represents record of 1 dataset verification.